lightweightsemaphore.h

```diff
@@ -0,0 +1,443 @@
+// Provides an efficient implementation of a semaphore (LightweightSemaphore)
+// - a synchronization primitive. This construct has been designed for 
```

Comment:
I think the "synchronization primitive" part can be left out. It should be fairly familiar to anyone searching for a semaphore...